SubArraySumEqualsK.py

```diff
@@ -0,0 +1,27 @@
+#Time Complexity : O(n))
+#Space Complexity : O(n) for dictionary
+#Did this code successfully run on Leetcode : yes
+#Any problem you faced while coding this : This one was tricky, but the lecture video helped me. I still don't fully understand
+#how to derive the 0:1 that we put inside the dictionary initially. I get why it's done,  but I don't know if i'd be able to come up with it
```

Comment:
Review the session again, it gives you a better understanding of how things work.
Sometimes the solution needs to be seen to remember it to be used in future. You'll get the hold of these approaches for further problems